daemon/graphdriver/lcow/lcow.go

```diff
@@ -21,158 +35,372 @@ import (
+	cachedSandboxMutex sync.Mutex                // Protects race conditions from multiple threads creating the cached sandbox.
+	cachedScratchFile  string                    // Location of the local cached empty scratch space.
+	cachedScratchMutex sync.Mutex                // Protects race conditions from multiple threads creating the cached scratch.
+	options            []string                  // Graphdriver options we are initialised with.
+	serviceVmsMutex    sync.Mutex                // Protects add/updates/delete to the serviceVMs map.
+	serviceVms         map[string]*serviceVMItem // Map of the configs representing the service VM(s) we are running.
+	globalMode         bool                      // Indicates if running in an unsafe/global service VM mode.
+
+	// NOTE: It is OK to use a cache here because Windows does not support
 	// restoring containers when the daemon dies.
 
-	// cacheMu is the mutex protection add/update/deletes to our cache
-	cacheMu sync.Mutex
-	// cache is the cache of all the IDs we've mounted/unmounted.
-	cache map[string]cacheType
+	cacheMutex sync.Mutex            // Protects add/update/deletes to cache.
+	cache      map[string]*cacheItem // Map holding a cache of all the IDs we've mounted/unmounted.
 }
 
-// InitLCOW returns a new LCOW storage driver.
-func InitLCOW(home string, options []string, uidMaps, gidMaps []idtools.IDMap) (graphdriver.Driver, error) {
+// deletefiles is a helper function for initialisation where we delete any
+// left-over scratch files in case we were previously forcibly terminated.
+func deletefiles(path string, f os.FileInfo, err error) error {
+	if strings.HasSuffix(f.Name(), ".vhdx") {
+		logrus.Warnf("lcowdriver: init: deleting stale scratch file %s", path)
+		return os.Remove(path)
+	}
+	return nil
+}
+
+// InitDriver returns a new LCOW storage driver.
+func InitDriver(dataRoot string, options []string, _, _ []idtools.IDMap) (graphdriver.Driver, error) {
 	title := "lcowdriver: init:"
-	logrus.Debugf("%s %s", title, home)
+
+	cd := filepath.Join(dataRoot, cacheDirectory)
+	sd := filepath.Join(dataRoot, scratchDirectory)
 
 	d := &Driver{
-		homeDir:             home,
-		options:             options,
-		cachedSandboxFile:   filepath.Join(home, "cache", sandboxFilename),
-		svmScratchSpaceFile: filepath.Join(home, "svmscratch", svmScratchFilename),
-		cache:               make(map[string]cacheType),
+		dataRoot:          dataRoot,
+		options:           options,
+		cachedSandboxFile: filepath.Join(cd, sandboxFilename),
+		cachedScratchFile: filepath.Join(cd, scratchFilename),
+		cache:             make(map[string]*cacheItem),
+		serviceVms:        make(map[string]*serviceVMItem),
+		globalMode:        false,
+	}
+
+	// Looks for relevant options
+	for _, v := range options {
+		opt := strings.SplitN(v, "=", 2)
+		if len(opt) == 2 {
+			switch strings.ToLower(opt[0]) {
+			case "lcowglobalmode":
```

Comment:
Also can you make this a boolean instead of `1` / `0`, and check the value here? `--storage-opt lcow.globalmode=false` should disable it. This is important, because we use these inside the `daemon.json` as well (for which we use booleans)